reward/supply_manager.go

```diff
@@ -379,9 +380,35 @@ func (sm *supplyManager) totalSupplyFromState(num uint64) (*big.Int, error) {
 	return totalSupply, nil
 }
 
+func (sm *supplyManager) getAccRewardUncached(num uint64) (*database.AccReward, error) {
+	accReward := sm.db.ReadAccReward(num)
+	if accReward != nil {
+		return accReward, nil
+	}
+
+	// Trace back to the last stored accumulated reward.
+	var fromNum uint64
+	var fromAcc *database.AccReward
+	for i := uint64(1); i < supplyReaccLimit; i++ {
```

Comment:
Why don't we just use `sm.db.ReadLastAccRewardBlockNumber`?

Reply:
LastAccRewardBlockNumber is _the highest_ number ever persisted. When API requested 257, we need to find 256 but Last... is like 161234567.